test/new-e2e/utils/e2e/client/agent.go

```diff
@@ -30,6 +35,56 @@ func (agent *Agent) init(auth *Authentification, data *agent.ClientData) error {
 	return err
 }
 
-func (agent *Agent) Status() (string, error) {
-	return agent.sshClient.Execute("sudo datadog-agent status")
+func (agent *Agent) Version() (string, error) {
+	return agent.sshClient.Execute("datadog-agent version")
+}
+
+type Status struct {
+	rawString string
+}
+
+func NewStatus(s string) *Status {
+	return &Status{rawString: s}
+}
+
+func (agent *Agent) Status() (*Status, error) {
+	s, err := agent.sshClient.Execute("sudo datadog-agent status")
+	if err != nil {
+		return nil, err
+	}
+	return NewStatus(s), nil
+}
+
+// IsReady true if status contains a valid version
+func (s *Status) IsReady() (bool, error) {
```

Comment:
I think `Status` shouldn't have a `IsReady` method and it would be the responsibility of `Status()` method to return a ready status. The reason is to avoid forcing the user to wait until the status is ready each time they call `Status` method.

Reply:
Changing this !